src/app/system/codes/view-code/view-code.component.ts

```diff
@@ -0,0 +1,215 @@
+      this.codeData = data.code;
+      this.codeValuesData = data.codevalues;
+    });
+  }
+
+  /**
+   * Creates and initializes the code values form.
+   */
+  ngOnInit() {
+    this.createCodeValuesForm();
+    this.initCodeValuesForm();
+  }
+
+  /**
+   * Initializes the code values form.
+   */
+  initCodeValuesForm() {
+    this.codeValuesData.forEach((codeValue: any) => {
+      this.codevalues.push(this.createCodeValuesRow(codeValue));
+      this.codeValueRowStatus.push('disabled');
+    });
+  }
+
+  /**
+   * Gets the code values form array.
+   * @returns {FormArray} Code values form array.
+   */
+  get codevalues(): FormArray {
+    return this.codeValuesForm.get('codevalues') as FormArray;
+  }
+
+  /**
+   * Creates the code values form.
+   */
+  createCodeValuesForm() {
+    this.codeValuesForm = this.formBuilder.group({
+      'codevalues': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Adds a row to the code values form.
+   */
+  addCodeValueRow() {
+    this.codevalues.push(this.createCodeValuesRow());
+    this.codevalues.at(this.codevalues.length - 1).enable();
+    this.codeValueRowStatus.push('new');
+  }
+
+  /**
+   * Creates a code value row in code values form.
+   * @param {string} codeValueName Code value name.
+   * @param {string} codeValueDescription Code value description.
+   * @param {number} codeValuePosition Code value position.
+   * @param {boolean} isCodeValueActive Code value is active or not.
+   */
+  createCodeValuesRow(codeValue?: any): FormGroup {
+    return this.formBuilder.group({
+      'name': [{ value: codeValue ? codeValue.name : '', disabled: true }, Validators.required],
+      'description': [{ value: codeValue ? codeValue.description : '', disabled: true }],
+      'position': [{ value: codeValue ? codeValue.position : 0, disabled: true }, Validators.required],
+      'active': [{ value: codeValue ? codeValue.active : false, disabled: true }]
+    });
+  }
+
+  /**
+   * Deletes the particular code value.
+   * @param {number} index Index of the row.
+   */
+  deleteCodeValue(index: number) {
+    const codeValueId = this.codeValuesData[index].id;
+    this.systemService.deleteCodeValue(this.codeData.id, codeValueId)
+      .subscribe((response: any) => { });
+    this.codeValuesData.splice(index, 1);
+    this.codevalues.removeAt(index);
+    this.codeValueRowStatus.splice(index, 1);
+  }
+
+  /**
+   * Removes/Deletes the particular code value which has not been intialised.
+   * @param {number} index Index of the row.
+   */
+  removeNewCodeValue(index: number) {
+    this.codevalues.removeAt(index);
+    this.codeValueRowStatus.splice(index, 1);
+  }
+
+  /**
+   * Updates the particular code value.
+   * @param {number} index Index of the row.
+   */
+  updateCodeValue(index: number) {
+    const updatedCodeValue: { name: string, description: string, position: number, isActive: boolean } = {
+      name: this.codevalues.at(index).get('name').value,
```

Comment:
Try to access this object from the array directly instead of constructing it here again.

Reply:
Can you please specify which array you are pointing to?

Reply:
`codevalues`